src/logger/src/error.rs

```diff
@@ -11,38 +11,53 @@ use std::fmt;
 pub enum LoggerError {
     /// First attempt at initialization failed.
     NeverInitialized(String),
-    /// The logger is locked while preinitializing.
-    IsPreinitializing,
     /// The logger is locked while initializing.
     IsInitializing,
     /// The logger does not allow reinitialization.
     AlreadyInitialized,
-    /// Writing to specified buffer failed.
-    LogWrite(std::io::Error),
-    /// Flushing messages stored in buffer to disk failed.
-    LogFlush(std::io::Error),
+    /// Writing or flushing the specified buffer failed.
+    Flush(std::io::Error),
+}
+
+/// Describes the errors which may occur while handling metrics scenarios.
+#[derive(Debug)]
+pub enum MetricsError {
```

Comment:
Since we are splitting the logger & metrics we can also separate the errors, or better put: we can define MetricError in metrics.rs and LoggerError in logger.rs instead of having a `error` module.

Reply:
Done.